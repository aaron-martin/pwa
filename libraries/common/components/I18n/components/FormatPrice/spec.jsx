/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import I18n from '../../index';

describe('<FormatPrice />', () => {
  let renderedElement;
  const testLocales = {
    greeting: 'Hello {price}',
  };
  const price = 1234.56;
  const formattedPrice = 'â‚¬1,234.56';
  const currency = 'EUR';
  const langCode = 'en-US';

  /**
   * Renders the component.
   * @param {Object} props The component props.
   */
  const renderComponent = (props) => {
    renderedElement = mount((
      <I18n.Provider {...props}>
        <div>
          <span className="only-price">
            <I18n.Price price={price} currency={currency} />
          </span>
          <span className="text-with-price">
            <I18n.Text string="greeting">
              <I18n.Price forKey="price" price={price} currency={currency} />
            </I18n.Text>
          </span>
        </div>
      </I18n.Provider>
    ));
  };

  beforeEach(() => {
    renderComponent({
      lang: langCode,
      locales: testLocales,
    });
  });

  describe('Given the component was mounted to the DOM', () => {
    it('should match snapshot', () => {
      expect(renderedElement).toMatchSnapshot();
    });

    it('should render formatted price', () => {
      const text = renderedElement.find('.only-price').text();
      expect(text).toBe(formattedPrice);
    });

    it('should render within translated text', () => {
      const text = renderedElement.find('.text-with-price').text();
      expect(text).toBe(`Hello ${formattedPrice}`);
    });
  });
});
