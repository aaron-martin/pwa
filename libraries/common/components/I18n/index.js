/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import I18nProvider from './components/I18nProvider';
import Placeholder from './components/Placeholder';
import Translate from './components/Translate';
import FormatDate from './components/FormatDate';
import FormatTime from './components/FormatTime';
import FormatPrice from './components/FormatPrice';

export default {
  Provider: I18nProvider,
  Text: Translate,
  Date: FormatDate,
  Time: FormatTime,
  Placeholder,
  Price: FormatPrice,
};
