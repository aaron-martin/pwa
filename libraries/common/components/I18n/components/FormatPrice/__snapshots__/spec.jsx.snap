// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormatPrice /> Given the component was mounted to the DOM should match snapshot 1`] = `
<I18nProvider
  lang="en-US"
  locales={
    Object {
      "greeting": "Hello {price}",
    }
  }
>
  <div>
    <span
      className="only-price"
    >
      <FormatPrice
        className=""
        currency="EUR"
        fractions={true}
        price={1234.56}
      >
        <span
          className=""
        >
          €1,234.56
        </span>
      </FormatPrice>
    </span>
    <span
      className="text-with-price"
    >
      <Translate
        className=""
        params={Object {}}
        string="greeting"
      >
        <span
          className=""
        >
          Hello 
          <FormatPrice
            className=""
            currency="EUR"
            forKey="price"
            fractions={true}
            key=".0"
            price={1234.56}
          >
            <span
              className=""
            >
              €1,234.56
            </span>
          </FormatPrice>
        </span>
      </Translate>
    </span>
  </div>
</I18nProvider>
`;
